opam-ci-check/bin/main.ml

```diff
@@ -218,20 +218,22 @@ let lower_bounds =
   in
   Arg.value (Arg.flag info)
 
-let distro_conv =
-  let of_string s =
-    match Dockerfile_opam.Distro.distro_of_tag s with
-    | Some _ -> Ok s (* We are just validating that it is a supported distro *)
-    | None -> Error (`Msg (s ^ " is not a supported distro"))
-  in
-  let pp = Fmt.string in
-  Arg.conv ~docv:"DISTRO" (of_string, pp)
-
-let arch_conv =
-  let of_string = Ocaml_version.arch_of_string in
-  let pp = Fmt.of_to_string Ocaml_version.string_of_arch in
-  Arg.conv ~docv:"ARCHITECTURE" (of_string, pp)
-
+let distro_options =
+  Dockerfile_opam.Distro.distros
+  |> List.map (fun distro ->
+      let tag = Dockerfile_opam.Distro.tag_of_distro distro in
+      (tag, tag))
+
+let arch_options =
+  Ocaml_version.arches
+  |> List.map (fun arch ->
+      (Ocaml_version.string_of_arch arch, arch))
+
+let arch_conv = Arg.enum arch_options
+let distro_conv = Arg.enum distro_options
+
+(* TODO: This should be limited to an enum, but we can't currently compute  it
+   without depending on /lib/build.ml *)
```

Comment:
As a follow-up we could also use the [compiler_variants function](https://github.com/ocurrent/ocaml-version/blob/f92a9f1b848b5eed179861ea13586a618b5fa25b/ocaml_version.mli#L578) (needs `arch` and compiler varsion) to validate the options provided, before trying to fetch Docker images for the specified configuration.